src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -58,17 +60,24 @@ object WdlExpression {
           case _ => throw new WdlExpressionException(s"Invalid operator: ${a.getName}")
         }
       case a: Ast if a.getName == "MemberAccess" =>
-        val lhs = evaluate(a.getAttribute("lhs"), lookup, functions).map {
-          case x: WdlObject => x
-          case _ => throw new WdlExpressionException("Left-hand side of expression must be a WdlObject")
-        }
         val rhs = a.getAttribute("rhs") match {
           case x:Terminal if x.getTerminalStr == "identifier" => x.getSourceString
           case _ => throw new WdlExpressionException("Right-hand side of expression must be identifier")
         }
-        lhs.map { _.value.getOrElse(rhs, throw new WdlExpressionException(s"Could not find key $rhs")) }
+        val lhs = evaluate(a.getAttribute("lhs"), lookup, functions).map {
```

Comment:
Would you rather have a function that's only ever called once?  Also once I fix the issue below, all cases become one liners, then it's just 3 lines!  The function that would have to be defined would have to be something like `lookupInObjectOrBinding(objectOrBinding: Object, str: String)`, or we'd have to figure out what we can call `WdlObject` and `WdlBinding` to make them have a common ancestor (`WdlObjectLike`?)  Not a bad idea, but also feels like it borders on the over-engineering considering we've only had to do this once (and it's hard to think of a time when you need to treat a namespace and a WdlObject the same except at expression evaluation time)